CareKit/CareKit/Shared/Task/Controller/OCKTaskController.swift

```diff
@@ -444,7 +444,13 @@ open class OCKTaskController: ObservableObject {
         let detailViewController = OCKDetailViewController(showsCloseButton: true)
         detailViewController.detailView.titleLabel.text = task.title
         detailViewController.detailView.bodyLabel.text = task.instructions
-        detailViewController.detailView.imageView.image = task.asset.map(UIImage.init(named:)) ?? nil
+        // We can't be sure if the image they provide is in the assets folder, in the bundle, or in a directory.
+        guard let asset = task.asset else { return detailViewController }
+        // We can check all 3 possibilities and then choose whichever is non-nil.
+        let symbol = UIImage(systemName: asset)
+        let appAssetsImage = UIImage(named: asset)
+        let otherUrlImage = UIImage(contentsOfFile: asset)
+        detailViewController.detailView.imageView.image = otherUrlImage ?? appAssetsImage ?? symbol
```

Comment:
I think we should extract this into a method on something in CareKit. We'll probably want to use it in other places as well. Perhaps it could live in a `UIImage+Asset.swift` file in the extensions directory.

```swift
extension UIImage {
    static func asset(_ name: String?) -> UIImage? { /*...*/ }
}
```

Reply:
I removed the example you showed in `OCKContactUtility` and replace the usage with the extension.